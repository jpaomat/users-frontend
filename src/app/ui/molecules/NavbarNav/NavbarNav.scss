.dropdown-close {
    .dropdown-menu {
        display: none;
    }
}