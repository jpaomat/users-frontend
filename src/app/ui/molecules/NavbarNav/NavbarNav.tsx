import { NavDropdown } from 'react-bootstrap';
import { NavbarLink, SimpleForm } from '../../../ui';
import { NavbarNavProps } from '../../../interfaces';
import './NavbarNav.scss';
import { useState } from 'react';

export const NavbarNav = ({ navbarItems }: NavbarNavProps) => {
	const [closeDropdown, setCloseDropdown] = useState<boolean>(false);
	
	return (
		<div className='collapse navbar-collapse' id='navbarNav'>
			<ul className='navbar-nav'>
				{navbarItems.map(({ label, redirectTo, dropdownForm }) =>
					dropdownForm ? (
						<NavDropdown
							key={label}
							className={`${closeDropdown ? 'dropdown-close' : 'show'}`}
							title={label}
							onClickCapture={( () => setCloseDropdown(false))}
						>
							<SimpleForm {...dropdownForm} onCloseForm={() => setCloseDropdown(true)} />
						</NavDropdown>
					) : (
						<NavbarLink key={label} label={label} redirectTo={redirectTo} />
					)
				)}
			</ul>
		</div>
	);
};
