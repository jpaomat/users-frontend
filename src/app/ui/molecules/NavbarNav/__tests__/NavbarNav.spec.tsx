import { render } from '@testing-library/react';
import { NavbarNav } from '../NavbarNav';
import { MemoryRouter } from 'react-router-dom';
import { defaultMock } from './mock';

describe('<NavbarNav /> component', () => {

	beforeEach(()=> jest.clearAllMocks);
	
	test('should render component', () => {
		const { container } = render(
			<MemoryRouter>
				<NavbarNav navbarItems={defaultMock}/>
			</MemoryRouter>
		);
		expect(container).toMatchSnapshot();
	});
});
