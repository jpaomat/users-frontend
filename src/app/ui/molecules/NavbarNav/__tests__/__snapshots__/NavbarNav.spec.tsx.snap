// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavbarNav /> component should render component 1`] = `
<div>
  <div
    class="collapse navbar-collapse"
    id="navbarNav"
  >
    <ul
      class="navbar-nav"
    >
      <a
        class="nav-link  undefined"
        href="/test"
      >
        test
      </a>
    </ul>
  </div>
</div>
`;
